lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -352,7 +348,8 @@ struct ExprVisitor {
       return {};
     }
     auto truncValue = value.as<uint64_t>().value();
-    return builder.create<moore::ConstantOp>(loc, type, truncValue);
+    return builder.create<moore::ConstantOp>(loc, cast<moore::IntType>(type),
```

Comment:
I have a question about why do we need to `cast<moore::IntType>(type)`? If I don't misunderstand, the `context.convertType()` will convert slang types like `ScalarType::Bit` into `!moore.i1`?

My working computer is encountering CPU over temperature error. So I can't immediately test these.

Reply:
Yes you're absolutely right, but `context.convertType()` also converts types like `string` into `!moore.string`. The constant op requires the type to be an `IntType` though (a simple bit vector). Slang guarantees that the type of the int literal is always an `IntType`, but `convertType()` doesn't know that.

Reply:
I get it.